How-to-upgrade-review-version.md

```diff
@@ -0,0 +1,399 @@
+that you won't have to change your build at all. You'll have the
+ability to get errors and completions in your editor and run `tsc`
+from the command line, but you won't have to integrate Typescript into
+your build.
+
+The way to do this is to not use Typescript. Stick with Javascript and
+use JSDoc to provide type information. This is less convenient than
+Typescript's syntax for types, but it means that your files stay
+boring old Javascript.
+
+Here's how. I'm going to use
+[typescript-eslint-parser](https://github.com/eslint/typescript-eslint-parser)
+as an example package so you can follow along if you want. Notably,
+this package *parses* Typescript, but is *written* in Javascript.
+
+## Add tsconfig
+
+I like to start with `tsc --init` and change the settings it gives you.
+
+I ended up with this:
+
+```json
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "commonjs",
+    "allowJs": true,
+    "checkJs": true,
+    "noEmit": true,
+    "resolveJsonModule": true
+    "esModuleInterop": true
+    "strict": false,
+  },
+  "exclude": [
+    "tests/fixtures",
+    "tests/integration"
+  ]
+}
+```
+
+For Javascript, your "compilerOptions" will pretty much always look
+like this. "resolveJsonModule" is optional, but you'll need it if your
+code ever `require`s a JSON file, so that Typescript will analyze it
+too. *Officially*, strict should be false, but as you will see later,
+ strict mode can be pretty great for Javascript code, not just
+ Typescript code.
+
+For typescript-eslint-parser, I added an exclude list. I wanted to
+check *all* the source files, because it's nice to have high
+aspirations. But it turns out the parser's tests take the form of JS
+files themselves, and I don't want to check those. Most of them are
+malformed in some way.
+
+You might want to use "include" if, say, you only want to check your
+source and not your tests or scripts. Or you can use "files" to give
+an explicit list of files to use, but this is annoying except for
+small projects.
+
+OK, you're all set. Run tsc and make sure it prints out errors. Now open up
+files in your editor and make sure the same errors show up there.
+
+Congratulations! You've done the only *required* part. Everything else
+just helps reduce the number of errors you see when editing.
+
+## Install @types packages.
+
+Your first order of business is to install types for packages you use.
+This is the easiest way to reduce the number of errors. Basically, if
+you use package X, and you see errors when you use it, try installing
+`@types/X`. Tons of packages have definitions, so you will probably
+find most of your dependencies are typed.
+
+Note that if you use node at all, you likely need `@types/node`.
+
+For typescript-eslint-parser, I installed:
+
+* @types/node
+* @types/jest
+* @types/estree
+* @types/shelljs
+* @types/eslint-scope
+
+If you are really, truly, completely in stealth mode, make sure not to
```

Comment:
You could request them in the tsconfig itself, no?

Reply:
Well, I didn't think so. Is that what Mine was referring to with the "types" compiler option?

Reply:
I believe so. Though I'm not sure 100% how it works - it may be language service only. It's not clear from the manual how these types are acquired.

https://www.typescriptlang.org/docs/handbook/tsconfig-json.html